packages/tinybased/src/lib/search/searched.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import {
+  create,
+  getByID,
+  insert,
+  insertBatch,
+  Lyra,
+  remove,
+  search,
+  SearchParams,
+  SearchResult,
+} from '@lyrasearch/lyra';
+
+import { ObjectToCellStringType, TableBuilder } from '../TableBuilder';
+import { OnlyStringKeys, TinyBaseSchema } from '../types';
+
+type LyraInstances<TSchema extends TinyBaseSchema> = {
+  [K in OnlyStringKeys<TSchema>]: Lyra<ObjectToCellStringType<TSchema[K]>>;
+};
+export class Searched<TSchema extends TinyBaseSchema = {}> {
+  public static ERRORS = {
+    NO_TABLES_DEFINED: 'No tables defined for search',
+    TABLE_ALREADY_DEFINED: (table: string) => `Table ${table} already defined.`,
+    INDEX_NOT_CONFIGURED: (index: string) =>
+      `Index ${index} not configured for search`,
+    TABLE_NOT_IN_SCHEMA_BUILDER: (tableKeyToIndex: string) =>
+      `Table ${tableKeyToIndex} cannot be indexed. It is not defined for the SchemaBuilder.`,
+  };
+
+  public readonly tables: Map<OnlyStringKeys<TSchema>, TableBuilder<any, any>> =
+    new Map();
+
+  private indexes: LyraInstances<TSchema> = {} as LyraInstances<TSchema>;
+
+  async initialize() {
+    if (this.tables.size === 0) throw Error(Searched.ERRORS.NO_TABLES_DEFINED);
+    for (const [key, table] of this.tables) {
+      const instance = await create({
+        schema: table.schema,
+      });
+      this.indexes = { ...this.indexes, [key]: instance };
+    }
+    return this;
```

Comment:
The pattern of being able to initialize multiple times, or to chain additional methods on after initialization feels strange.

I'd prefer if we followed the existing patterns established in this repo for separating out a `Builder` class versus the thing that the user actually interacts with. Calling `initialize` would then return a new object instead of `this`.